src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -159,17 +160,40 @@ class IOUAmountPage extends React.Component {
      * Update amount on amount change
      * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit
      *
-     * @param {String} amount
+     * @param {String} text Changed text from user input
      */
-    updateAmount(amount) {
-        if (!this.validateAmount(amount)) {
-            return;
-        }
+    updateAmount(text) {
+        this.setState((prevState) => {
+            const amount = this.replaceAllDigits(text, this.props.fromLocaleDigit);
+            return this.validateAmount(amount)
+                ? {amount: this.stripCommaFromAmount(amount)}
+                : prevState;
+        });
+    }
 
-        this.setState({amount: this.stripCommaFromAmount(amount)});
+    /**
+     * Replaces each character by calling `convertFn`. If `convertFn` throws an error, then
+     * the original character will be preserved.
+     *
+     * @param {String} text
+     * @param {Function} convertFn `this.props.fromLocaleDigit` or `this.props.toLocaleDigit`
+     * @returns {String}
+     */
+    replaceAllDigits(text, convertFn) {
+        return _.chain([...text])
+            .map((char) => {
+                try {
+                    return convertFn(char);
+                } catch {
+                    return char;
+                }
+            })
+            .join('')
+            .value();
     }
 
     render() {
+        const formattedAmount = this.replaceAllDigits(this.state.amount, this.props.toLocaleDigit);
```

Comment:
Can't we just use, `numberFormat` prop directly for the whole input value?

Reply:
Unfortunately we can't. Because the amount might not be a valid number yet while inputing (e.g. "12.").

If we do need to use numberFormat the code will be 

```
get formattedInputingAmount() {
        if (!this.state.amount) {
            return '';
        }

        const endsWithDecimalPoint = this.state.amount.endsWith('.');
        if (endsWithDecimalPoint) {
            const localized = this.props.numberFormat(this.state.amount, {useGrouping: false, minimumFractionDigits: 1});
            return localized.slice(0, -1);
        }

        const fraction = this.state.amount.split('.')[1];
        const fractionDigits = fraction ? fraction.length : 0;
        return this.props.numberFormat(this.state.amount, {useGrouping: false, minimumFractionDigits: fractionDigits});
    }
```

Reply:
Ok. What is more performant?

Reply:
There is no performance issue here since the max length of `amount` is just 8. But `replaceAllDigits` here is more readable.

Reply:
Sounds, good. I will give it a go.